### ISSUE TITLE
Environment variables, LabRAD registry and Python package paths 

### ISSUE THREAD

**Issue by patzinak** [McDermott-Group/servers#74]
I see we our repos are getting more messy and messy. It would be good to established some specific practices.

dataChest uses the environment variable to tell Python about its location. My measurement scripts do this dynamically in an albeit very ugly way. But this does not require the user to configure anything.

``` Python
if __file__ in [f for f in os.listdir('.') if os.path.isfile(f)]:
    SCRIPT_PATH = os.path.dirname(os.getcwd())
else:
    SCRIPT_PATH = os.path.dirname(__file__)
LOCAL_PATH = SCRIPT_PATH.rsplit('measurement', 1)[0]
MEASUREMENT_PATH = os.path.join(LOCAL_PATH, 'measurement')
if MEASUREMENT_PATH not in sys.path:
    sys.path.append(MEASUREMENT_PATH) 
```

We also have untilites.py duplicated in two different repository locations. It seems like it would be reasonable to have a part of the server repo to act as a Python package and have only a single environment variable name assign to it. This will allow importing dataChest, utilities.py and anything else from anywhere. I don't think it is a good idea to have several paths defined by the environment variables because the number of the paths will likely expand it time. The other reason why this is important is that I've modified the experiment.py to allow for saving data at the end of the sweeps. However, I assumed that the dataChest has been configured properly on all machines. If this is not the case, people will see errors. I have to modify the installation instructions now to refer to the .pdf dataChest installation instructions. I will do this. This, however, wouldn't be maintainable in the long run because the entire software environment becomes more and more inhomogeneous.

Similarly, dataChest uses an environment variable for the root data path while @roboguy222 and me are using the LabRAD registry for the similar purposes. @amopremcak it is definitely fine to keep the things the way they are now, but it would be great if the LabRAD wrapper for the dataChest will search in the LabRAD registry for the proper path (somewhere in ['Local', 'computer-name', 'dataChest Path']), and if it can't find the path in the registry you can let it to search among the environment variables. Something like this.

The other inconsistency is the usage of .docx (not really applicable to us), .pdf and .md files for the documentation. One can also start using the wiki. My personal opinion is to keep the information as close to the code as possible and try to put as much info in folder-specific readme.md files or manual.md files. They are easy to edit and update. And they either exist in the folder or not, you don't need to guess whether anything is written about that part of the code.

This is just something for everybody to think about. I'm pretty sure if similar problems are solved in similar ways this will help to maintain the project. On the other hand, I'm fine with everything as it is because everything works well for me now. This is because I have some general understanding but I think all this is not that transparent to the newcomers. I want to let you guys decide if anything should be done about this.

**Comment by roboguy222** [comment]
About the documentation: The reason Alex has it as a pdf here is because he has equations in the documentation, and apparently github wiki's do not support them.  I doubt you can do them in .md files either.

**Comment by amopremcak** [comment]
What utilites.py file is duplicated in two locations? Also, what other libraries exist in the servers repository that we would want to import from other locations on a lab machine?

**Comment by patzinak** [comment]
Yes, utilitis.py is in both folders, gpibdevices and serial devices. I'm not aware about any other libraries at the moment though @nmGit has developed a GUI suit but I'm not sure how the importing works. I see `import MGui` in one of his files but I don't see in his manual anything about adding the Python path. If@nmGit if you added a path to the environment variable could you please add a comment on this to your readme.md?

Yes, I'm aware about the problem with the equations (although not sure whether they are really that important, seems like some of them could be skipped or duck-typed here). Anyway, it would be great to have .tex file here so people can modify it. For example, I wanted to put a note that you can not yet use `addParameter` on variables with units, i.e. of `labrad.units.Value` type. I was under impression that it was implemented but I understand now that this will be in the wrapper that is not yet written.

**Comment by nmGit** [comment]
@patzinak I added a couple lines asking users to refer to the datachest documentation for datachest setup as I did not want to duplicate information. I also just added the steps to setup the python path
